rw_tlv.m4

```diff
@@ -355,131 +355,71 @@ m4_define(['m4_into_fields'], ['{m4_into_fields_lhs(['$2'], [''], $1_FIELDS)['}
 
 // Reduction macro.
 // Performs an operation across all instances of a hierarchy and provides the result outside that hierarchy.
-// m4+redux($sum[7:0], >hier, max, min, $addend, '0, +)
-m4_define_plus(['m4_redux'], ['
-
-m4_pushdef(['m4_redux_sig'],    ['$5'])
-m4_pushdef(['m4_hier'],         ['$6'])
-m4_pushdef(['m4_MAX'],          ['$7'])
-m4_pushdef(['m4_MIN'],          ['$8'])
-m4_pushdef(['m4_sig'],          ['$9'])
-m4_pushdef(['m4_init'],         ['$10'])
-m4_pushdef(['m4_op'],           ['$11'])
-
-'], m4___file__, m4___line__, ['
+// m4+redux($sum[7:0], /hier, max, min, $addend, '0, +)
+\TLV redux($_redux_sig,/_hier,#_MAX,#_MIN,$_sig,$_init,#_op)
    \always_comb
-      $m4_redux_sig = m4_init;
-      for (int i = m4_MIN; i <= m4_MAX; i++)
-         m4_redux_sig = m4_redux_sig m4_op m4_hier[i]m4_sig;
-'],
-
-['
-m4_popdef(['m4_redux_sig'])
-m4_popdef(['m4_hier'])
-m4_popdef(['m4_MAX'])
-m4_popdef(['m4_MIN'])
-m4_popdef(['m4_sig'])
-m4_popdef(['m4_init'])
-m4_popdef(['m4_op'])
-'])
+      $_redux_sig = $_init;
+      for (int i = #_MIN; i <= #_MAX; i++)
+         $_redux_sig = $_redux_sig #_op /_hier[i]$_sig;
 
 
 // Similar to m4+redux, but each element is conditioned.
 // Performs an operation across all instances of a hierarchy and provides the result outside that hierarchy.
-// m4+redux_cond($selected_value[7:0], >hier, max, min, $value, '0, |, $select)
-m4_define_plus(['m4_redux_cond'], ['
-
-m4_pushdef(['m4_redux_sig'],    ['$5'])
-m4_pushdef(['m4_hier'],         ['$6'])
-m4_pushdef(['m4_MAX'],          ['$7'])
-m4_pushdef(['m4_MIN'],          ['$8'])
-m4_pushdef(['m4_sig'],          ['$9'])
-m4_pushdef(['m4_init'],         ['$10'])
-m4_pushdef(['m4_op'],           ['$11'])
-m4_pushdef(['m4_cond_expr'],    ['$12'])
-
-'], m4___file__, m4___line__, ['
-   m4_hier[*]
-      m4_sig['']_cond = m4_cond_expr ? m4_sig : m4_init;
+// m4+redux_cond($selected_value[7:0], /hier, max, min, $value, '0, |, $select)
+\TLV redux_cond($_redux_sig,/_hier,#_MAX,#_MIN,$_sig,$_init,#_op,$_cond_expr)
+   /_hier[*]
+      $_sig['']_cond = $_cond_expr ? $_sig : $_init;
    \always_comb
-      $m4_redux_sig = m4_init;
-      for (int i = m4_MIN; i <= m4_MAX; i++)
-         m4_redux_sig = m4_redux_sig m4_op m4_hier[i]m4_sig;
-'],
-
-['
-m4_popdef(['m4_redux_sig'])
-m4_popdef(['m4_hier'])
-m4_popdef(['m4_MAX'])
-m4_popdef(['m4_MIN'])
-m4_popdef(['m4_sig'])
-m4_popdef(['m4_init'])
-m4_popdef(['m4_op'])
-m4_popdef(['m4_cond_expr'])
-'])
+      $_redux_sig = $_init;
+      for (int i = #_MIN; i <= #_MAX; i++)
+         $_redux_sig = $_redux_sig #_op /_hier[i]$_sig;
 
 
 // Select across a hierarchy (MUX) within a pipeline with a decoded select.  Works for $pipe_signals and $ANY.
-// m4+select($selected_value[7:0], >top, >hier, ...fix)
```

Comment:
This is used by `vc_flop_fifo_v2` which is used by the grid_network. Currently, that example is broken because of the 1b syntax in `m4+select`. We'll debug grid_network to debug this macro, but we can push these changes prior to debugging.